lib/streaming-pipeline.js

```diff
@@ -6,23 +6,15 @@ const logger = require('util').debuglog('riff');
 const InputUnmarshaller = require('./input-unmarshaller');
 const {guardWithTimeout} = require('./async');
 
-const toObject = (parameters) => {
+const toObject = (parameters, streamNames) => {
     const result = {};
     for (let i = 0; i < parameters.length; i++) {
-        result[`${i}`] = parameters[i];
+        const stream = parameters[i];
+        result[`${i}`] = stream;
```

Comment:
Object keys are always strings, they will automatically be coerced

```suggestion
        result[i] = stream;
```

Reply:
Hmm, I find it a bit hard to guess for moderate JS users (like me).